### PR TITLE
Deployments code cleanup

### PR DESCRIPTION
**Pull request by andrewazores**
These commits simply clean up the Deployments area. For the most part it's just simple indentation type fixes, but there are also a fair number of type annotations added where previously either no type was declared or the type was declared as "any". There is also one larger commit ("add PodPhase enum type") that adds a new enum type to encapsulate the running states of deployment pods and reduce duplication of strings that were previously used.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2528-1 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2528-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2528-2 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2528-fabric8-ui.badger.fabric8.io

File: src/app/space/create/deployments/resource-usage/deployments-resource-usage.component.ts [hunk 1]
```diff
@@ -10,10 +10,7 @@ import { Environment } from '../models/environment';
   templateUrl: 'deployments-resource-usage.component.html'
 })
 export class DeploymentsResourceUsageComponent {
-
   @Input() environments: Observable<Environment[]>;
   @Input() spaceId: Observable<string>;
-
-  constructor() { }
-
+  resourcesCollapsed: boolean = false;
```
Comment (jiekang) [inline src/app/space/create/deployments/resource-usage/deployments-resource-usage.component.ts]: I'd prefer a newline between annotated variables and ones that aren't annotated.

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2528-3 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2528-fabric8-ui.badger.fabric8.io

File: src/app/space/create/deployments/apps/deployment-card-container.component.spec.ts [hunk 2]
```diff
@@ -26,31 +34,35 @@ class FakeDeploymentCardComponent {
...
+    },
+    (component: DeploymentCardContainerComponent) => {
       component.spaceId = 'space';
       component.environments = Observable.of(environments);
       component.application = 'app';
     });
 
-  it('should created children components with proper objects', function(this: Context) {
-    let arrayOfComponents = this.fixture.debugElement.queryAll(By.directive(FakeDeploymentCardComponent));
+  it('should creat child components with proper inputs', function(this: Context) {
```
Comment (jiekang) [inline src/app/space/create/deployments/apps/deployment-card-container.component.spec.ts]: s/create/create

File: src/app/space/create/deployments/apps/deployment-status-icon.component.ts [hunk 3]
```diff
@@ -19,15 +19,13 @@ enum CLASSES {
...
 
 const STAT_THRESHOLD = .6;
+
 @Component({
   selector: 'deployment-status-icon',
   templateUrl: 'deployment-status-icon.component.html'
 })
 export class DeploymentStatusIconComponent {
-
+  static readonly CLASSES = CLASSES;
```
Comment (jiekang) [inline src/app/space/create/deployments/apps/deployment-status-icon.component.ts]: Prefer a space between this and the annotated variables.

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2528-4 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2528-fabric8-ui.badger.fabric8.io

**Comment by jiekang** [comment]
@andrewazores any particular reason for the use of the exported class versus the file name?

`refactor(deployments): refactor UtilizationBarComponent`
`style(deployments): clean up DeploymentCardContainer code style and types`

`refactor(deployments): clean up deployments-resource-usage component code style`

Maybe pick one? I actually prefer the file version since they may export other things which have been refactored

Well; this is super nitpicky... feel free to ignore.

**Comment by andrewazores** [comment]
@jiekang not sure why I switched or what reasons I had... I'll reword them all to use the file name, since you're right, some of them do export not just one major Component/Service but also a type or something, which may have been changed independently of the component/service.

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2528-5 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2528-fabric8-ui.badger.fabric8.io

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2528-6 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2528-fabric8-ui.badger.fabric8.io

File: src/app/space/create/deployments/apps/deployment-card.component.spec.ts [hunk 4]
```diff
@@ -3,35 +3,44 @@ import {
   DebugElement,
   Input
 } from '@angular/core';
+
```
Comment (joshuawilson) [inline src/app/space/create/deployments/apps/deployment-card.component.spec.ts]: The addition of the new lines between imports effects the linter. The imports are grouped alphabetically by the library name, not the imported component. The linter only looks at imports that have no new lines. 

The layout that I tried to use was Angular in the first group, then external, then internal. On very large imports I might break it down further.

On a different note, if you don't want the linter to check the order of these we can have that discussion. But I see no value in the new lines.

**Comment by fabric8cd** [comment]
@andrewazores fabric8/fabric8-ui:SNAPSHOT-PR-2528-7 fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2528-fabric8-ui.badger.fabric8.io

**Pull request merged**